src/algorithm/simplifyvw.rs

```diff
@@ -173,36 +437,165 @@ pub trait SimplifyVW<T, Epsilon = T> {
+    ///
+    /// **Note**: if removal of a point causes a self-intersection, but the geometry only has `n + 2`
+    /// points remaining (4 for a `LineString`, 6 for a `Polygon`), the point is retained and the
+    /// simplification process ends. This is because there is no guarantee that removal of two points will remove
+    /// the intersection, but removal of further points would leave too few points to form a valid geometry.
+    ///
+    /// ```
+    /// use geo::{Point, LineString};
+    /// use geo::algorithm::simplifyvw::{SimplifyVWPreserve};
+    ///
+    /// let mut vec = Vec::new();
+    /// vec.push(Point::new(10., 60.));
+    /// vec.push(Point::new(135., 68.));
+    /// vec.push(Point::new(94., 48.));
+    /// vec.push(Point::new(126., 31.));
+    /// vec.push(Point::new(280., 19.));
+    /// vec.push(Point::new(117., 48.));
+    /// vec.push(Point::new(300., 40.));
+    /// vec.push(Point::new(301., 10.));
+    /// let linestring = LineString(vec);
+    /// let mut compare = Vec::new();
+    /// compare.push(Point::new(10., 60.));
+    /// compare.push(Point::new(126., 31.));
+    /// compare.push(Point::new(280., 19.));
+    /// compare.push(Point::new(117., 48.));
+    /// compare.push(Point::new(300., 40.));
+    /// compare.push(Point::new(301., 10.));
+    /// let ls_compare = LineString(compare);
+    /// let simplified = linestring.simplifyvw_preserve(&668.6);
+    /// assert_eq!(simplified, ls_compare)
+    /// ```
+    fn simplifyvw_preserve(&self, epsilon: &T) -> Self
+    where
+        T: Float + SpadeFloat;
+}
+
+impl<T> SimplifyVWPreserve<T> for LineString<T>
+where
+    T: Float + SpadeFloat,
+{
+    fn simplifyvw_preserve(&self, epsilon: &T) -> LineString<T> {
+        let gt = GeomSettings {
+            initial_min: 2,
+            min_points: 4,
+            geomtype: GeomType::Line,
+        };
+        let mut simplified = vwp_wrapper(&gt, &self.0, None, epsilon);
+        LineString(simplified.pop().unwrap())
+    }
+}
+
+impl<T> SimplifyVWPreserve<T> for MultiLineString<T>
+where
+    T: Float + SpadeFloat,
+{
+    fn simplifyvw_preserve(&self, epsilon: &T) -> MultiLineString<T> {
+        MultiLineString(
+            self.0
+                .iter()
+                .map(|l| l.simplifyvw_preserve(epsilon))
+                .collect(),
+        )
+    }
+}
+
+impl<T> SimplifyVWPreserve<T> for Polygon<T>
+where
+    T: Float + SpadeFloat,
+{
+    fn simplifyvw_preserve(&self, epsilon: &T) -> Polygon<T> {
+        let gt = GeomSettings {
+            initial_min: 4,
+            min_points: 6,
+            geomtype: GeomType::Ring,
+        };
+        let mut simplified = vwp_wrapper(&gt, &self.exterior.0, Some(&self.interiors), epsilon);
+        let exterior = LineString(simplified.remove(0));
+        let interiors = match simplified.is_empty() {
+            true => vec![],
+            false => simplified.into_iter().map(LineString).collect(),
+        };
```

Comment:
seems fine to just do `let interiors = simplified.into_iter().map(LineString).collect();` here. an empty iterator will result in an empty `Vec`